File: ΜΑΘΗΜΑ_9/UML-JAVA/department.java
import java.util.*

class employee
{
    private String name;
    private int age;
    private double salary;
    public employee(String n,int a,double s)
    {
        this.name=n;
        this.age=a;
        this.salary=s;
    }
    public String getName() {return this.name;}
    public int getAge() {return this.age;}
    public double getSalary() {return this.salary;}
    public void info()
    {
        System.out.println(this.name+"-"+String.valueOf(this.age)+"-"+String.valueOf(this.salary));
    }
    public String toString()
    {
        return this.name+"-"+String.valueOf(this.age)+"-"+String.valueOf(this.salary);
    }
}

class department
{
   private String id;
   private List <employee> emps;
   public employee(String id)
   {
       this.id;
       this.emps=new ArrayList<employee>();
   }
   public void add_Employee(employee e)
   {
      this.emps.add(e);
   }
   public void Print_Employees()
   {
      for(employee e:this.emps)
      {
          System.out.println(e);
      }
   }
}

File: 2020-2021/ΜΑΘΗΜΑ_9/UML-C++/department.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
#include <ostream>
class employee
{
    std::string name;
    int age;
    double salary;
    public:
      employee(std::string n,int a,double s):name(n),age(a),salary(s) 
      {}
      std::string getName()const {return this->name;}
      int getAge()const {return this->age;}
      double getSalary()const {return this->salary;}
      void info()
      {
         std::cout<<this->name<<"-"<<this->age<<"-"
          <<this->salary<<std::endl;
      }
      friend std::ostream &operator<<(std::ostream &os,const employee &e)
      {
          return os<<e.name<<"-"<<e.age<<"-"<<e.salary<<std::endl;
      }
};

class department
{
    private:
      int id;
      std::vector <employee> emps;
    public:
      department(int i):id(i) {}
      void add_employee(const employee &e)
      {
          this->emps.push_back(e);
      }
      int getId()const {return this->id;}
      void Print_Employees()
      {
         std::ostream_iterator <employee> out(std::cout,"\n");
         std::copy(this->emps.begin(),this->emps.end(),out);         
      }
};